packages/sdk/src/Actions/EventAction.ts

```diff
@@ -472,6 +477,36 @@ export class EventAction extends DeployableTarget<
    * @returns {Promise<boolean>}
    */
   public async isActionStepValid(
+    actionStep: ActionStep,
+    params?: GetLogsParams<Abi, ContractEventName<Abi>> & {
```

Comment:
i think instead of this awkward type, it should be something more like 
```
export type ValidateEventStepParams = Omit<
  GetLogsParameters<Abi, ContractEventName<Abi>> & {
    logs?: Log[];
    knownEvents?: Record<Hex, AbiEvent>;
  },
  'address'
>;

export type ValidateFunctionStepParams = GetTransactionParameters;

params?: ValidateEventStepParams &
  ValidateFunctionStepParams & { chainId?: number };
```

and you can use this type in all the validation methods, and pass in this object to both `getLogs` and `getTransaction`